kratos/geometries/line_2d_2.h

```diff
@@ -883,6 +884,31 @@ class Line2D2 : public Geometry<TPointType>
         const double Tolerance = std::numeric_limits<double>::epsilon()
         ) override
     {
+        // The center of the geometry
+        const auto& r_center = this->Center();
+
+        // We compute the normal to check the normal distances between the point and the triangles, so we can discard that is on the triangle
+        const CoordinatesArrayType aux_point = ZeroVector(3);
+        const array_1d<double, 3> normal = this->UnitNormal(aux_point);
+
+        // We compute the distance, if it is not in the pane we
+        const Point point_to_project(rPoint);
+        Point point_projected;
+        const array_1d<double,3> vector_points = rPoint - r_center.Coordinates();
+
+        const double distance = GeometricalProjectionUtilities::FastProjectDirection(*this, point_to_project, point_projected, normal,  vector_points, 0);
+
+        // We check if we are on the plane
+        if (std::abs(distance) > std::numeric_limits<double>::epsilon()) {
+            if (std::abs(distance) > 1.0e-6 * Length()) {
+                KRATOS_WARNING("Line2D2") << "The point of coordinates X: " << rPoint[0] << "\tY: " << rPoint[1] << " it is in a distance: " << std::abs(distance) << std::endl;
+                return false;
+            }
+
+            // Not in the plane, but allowing certain distance, projecting
+            noalias(point_projected) = rPoint - normal * distance;
+        }
+
```

Comment:
Is there any test for this? I think the implementation is not correct.

Reply:
Yes, the current tests for IsInside. It is doing the same for the triangle

Reply:
But this is a line_2d_2

Reply:
Is the same than the triangle in 3D, it follows the same principles

Reply:
No, I disagree, the implmentation can be simplified and seems to be a copy paste of triangle. I suggest to revise it and make test for it.

Reply:
I recommend GeometricalProjectionUtilities::FastProjectOnLine, does exactly that

Reply:
Done

Reply:
Why the warning? Consider that we will run this methods millions of times!